mesh/node.c

```diff
@@ -2312,6 +2312,43 @@ static bool ivindex_getter(struct l_dbus *dbus, struct l_dbus_message *msg,
 	return true;
 }
 
+static struct l_dbus_message *seq_nr_setter(struct l_dbus *dbus,
+					struct l_dbus_message *msg,
+					struct l_dbus_message_iter *value,
+					l_dbus_property_complete_cb_t
+								complete_cb,
+					void *user_data)
+{
+	struct mesh_node *node = user_data;
+	struct mesh_net *net = node_get_net(node);
+	uint16_t seq_nr;
+
+	l_debug("seq_nr setter");
+
+	if (l_dbus_message_iter_next_entry(value, &seq_nr)) {
+		l_info("Sequence number to be set: %u", seq_nr);
+		mesh_net_set_seq_num(net, seq_nr);
```

Comment:
What if we've just increased `seq_num` beyond requested value?

Since setters on `DBus.Properties' don't seem to return anything, maybe it would be better to add an explicit method for this, and maybe even make it work like atomics (compare/exchange) in C?

Reply:
Setter property setter changed into method. Value check still need to be verified